### PR TITLE
[5.1] Add Tag Relationship in ContentSerializer.php

### PR DESCRIPTION
**Pull request by alexandreelise**
The work was already done by Joomla Team and other contributors what I did is just to use the TagApiSerializerTrait in ContentSerializer. It should be added everywhere where the tags are supported but to make it simple I added it just to ContentSerializer. Hope it helps. Tested it locally on my setup with Postman and it works.

Pull Request for Issue # .
No issues for this pull request

### Summary of Changes
Add the Tag Relationship for ContentSerializer

### Testing Instructions
1. Apply the patch
2. Run this command   curl -X GET -H "X-Joomla-Token: YOUR_JOOMLA_API_TOKEN" -H "Accept: application/vnd.api+json" --url "https://example.org/api/index.php/v1/content/articles"
3. It should return a response in JSON:API format containing the tags relationship

### Actual result BEFORE applying this Pull Request
The tag relationship does not appear when doing a GET request on /api/index.php/v1/content/articles 

### Expected result AFTER applying this Pull Request
The tag relationship appears successfully alongside the other relationships like category and created_by

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [ ] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Review by alexandreelise — COMMENTED** [review]
csfixer does not complain locally in PHPStorm but maybe GitHub Web Editor changes stuff when I paste the content of the local file on GitHub directly

**Review by alexandreelise — COMMENTED** [review]
comply to phpcs rules that make drone phpcs task fail for now. I reproduced the exact same command as the drone phpcs task does and I pass locally. Now it should work.

**Comment by alexandreelise** [comment]
Thx for your feedback @richard67

**Review by wilsonge — APPROVED** [review]
Super obvious on review that I just missed this on oversight when adding the trait in https://github.com/joomla/joomla-cms/commit/3b82bf129322cf473f10a1856d4ec385895ff1ad

**Comment by wilsonge** [comment]
@laoneo @MacJoom you probably want this backported into 4.4

**Comment by alexandreelise** [comment]
Thanks for this Trait @wilsonge

**Comment by HLeithner** [comment]
if it's also an issue in 4.4 it have to be recreated against 4.4 branch and upmerged later. can you please do this @alexandreelise ?

**Comment by richard67** [comment]
@alexandreelise Upmerge means later when your PR for 4.4-dev will be done and merged into 4.4-dev by a maintainer, maintainers or release leads later merge that change from 4.4-dev up into 5.0-dev (and 5.1-dev) so that fix will be available with both 4.4.1 and 5.0.1 (if ready in time). Nothing you should worry about, you only need to knw that you should redo this PR for the 4.4-dev branch.

**Comment by alexandreelise** [comment]
> @alexandreelise Upmerge means later when your PR for 4.4-dev will be done and merged into 4.4-dev by a maintainer, maintainers or release leads later merge that change from 4.4-dev up into 5.0-dev (and 5.1-dev) so that fix will be available with both 4.4.1 and 5.0.1 (if ready in time). Nothing you should worry about, you only need to knw that you should redo this PR for the 4.4-dev branch.

Thanks for clarifying @richard67

**Comment by alexandreelise** [comment]
Hi. Should I update this branch as suggested by GitHub to be aligned with upstream? @richard67

**Comment by alexandreelise** [comment]
> if it's also an issue in 4.4 it have to be recreated against 4.4 branch and upmerged later. can you please do this @alexandreelise ?

@HLeithner PR done same as in 5.0-dev as suggested. https://github.com/joomla/joomla-cms/pull/42257

**Comment by richard67** [comment]
> Hi. Should I update this branch as suggested by GitHub to be aligned with upstream? @richard67

@alexandreelise No, you can close this PR here as it is the same as the new PR for 4.4-dev so it doesn't need an extra one for 5.0-dev.

**Comment by alexandreelise** [comment]
Ok thanks

**Comment by richard67** [comment]
Closing in favour of #42257 .

**Comment by alexandreelise** [comment]
> > Hi. Should I update this branch as suggested by GitHub to be aligned with upstream? @richard67
> 
> @alexandreelise No, you can close this PR here as it is the same as the new PR for 4.4-dev so it doesn't need an extra one for 5.0-dev.

One more thing. Should my future pull request favor oldest supported joomla cms branch so that other maintainers or release lead can upmerge them easier than to redo this mistake I did today? @richard67

**Comment by richard67** [comment]
@alexandreelise Bug fixes for bugs which exist in 4.4-dev and 5.0-dev have to be made for the 4.4-dev branch. Bug fixes for bugs which only exist in 5.x have to be made for the 5.0-dev branch. New features have to be made for the 5.1-dev branch.

**Comment by alexandreelise** [comment]
> @alexandreelise Bug fixes for bugs which exist in 4.4-dev and 5.0-dev have to be made for the 4.4-dev branch. Bug fixes for bugs which only exist in 5.x have to be made for the 5.0-dev branch. New features have to be made for the 5.1-dev branch.

Thanks for clarifying. Now I understand the process. Have a delightful day @richard67

**Comment by HLeithner** [comment]
reopen and rebase explained in https://github.com/joomla/joomla-cms/pull/42257#issuecomment-1788660599

**Comment by LadySolveig** [comment]
Many thanks for this very nice PR @alexandreelise !

**Comment by alexandreelise** [comment]
I'm always happy to help @LadySolveig

**Comment by alexandreelise** [comment]
hurray celebrate! super joomlers around the world! we love joomla!

File: api/components/com_content/src/Serializer/ContentSerializer.php [hunk 1]
```diff
@@ -4,13 +4,14 @@
  * Joomla! Content Management System
  *
  * @copyright  (C) 2019 Open Source Matters, Inc. <https://www.joomla.org>
- * @license    GNU General Public License version 2 or later; see LICENSE.txt
+ * @license        GNU General Public License version 2 or later; see LICENSE.txt
```
Comment (laoneo) [inline api/components/com_content/src/Serializer/ContentSerializer.php:7]: Can you make a new pr which reverts this change here?

**Comment by alexandreelise** [comment]
Someone is ruining the party. We should celebrate! A random contributor contributes to our beloved project joomla. We love joomla. Bye...for now.

**Comment by laoneo** [comment]
Do you mean me?

**Comment by alexandreelise** [comment]
Just to clarify. Why did CI pass if this PR is wrong why was it merged @laoneo ?

**Comment by laoneo** [comment]
I did not say that the pr is wrong. Only that particular line. And CI does not check the doc blocks. They were brought to this good state by many contributors in all these years.

Why was it merged? Here and there are pr's merged where some stuff was wrong and then we fix it afterwards. This is normally no problem and contributors are happy to help. If you can't, I will do it by myself, no problem.

File: api/components/com_content/src/Serializer/ContentSerializer.php (hunk 1 above)
Comment (laoneo) [reply api/components/com_content/src/Serializer/ContentSerializer.php:7]: Solved by #42278.

Comment (LadySolveig) [reply api/components/com_content/src/Serializer/ContentSerializer.php:7]: Thank you for paying attention on this @laoneo PR to revert the change #42278

**Comment by alexandreelise** [comment]
Have a delightful day @laoneo. Thanks for your help.

**Comment by laoneo** [comment]
You don't have to thank me, more @LadySolveig which fixed it while some other people where discussing it.

**Pull request merged**